ofxProjectGenerator/src/projects/xcodeProject.cpp

```diff
@@ -128,10 +128,12 @@ bool xcodeProject::createProjectFile(){
 
 	// make everything relative the right way.
 	string relRoot = getOFRelPathFS(projectDir).string();
-
+	projectDir = projectDir.lexically_normal();
+	if (projectDir.is_absolute()) {
+		relRoot = getOFRoot();
```

Comment:
my experience is getOFRoot() is always absolute - but might be worth checking.

Reply:
I think it is, yes. in this case I'm setting relRoot to absolute root path, so we don't have some absurd relative paths going all the way down to root, like
../../../../../../../Users/z/Desktop/project

Reply:
I think you might need a bit more handling here.  First you would want to make `relRoot` relative to projectDir but  only make relative if the projectDir exists inside the OF folder. 

I think very similar to what I did here: 
https://github.com/openframeworks/projectGenerator/blob/master/ofxProjectGenerator/src/projects/baseProject.cpp#L221-L224

Reply:
obviously in this case relRoot is misnamed. in fact updating PG always have the feeling it has to be completely rewritten :)

Reply:
Otherwise I think it will end up replacing ../../../ with absolute OF paths :)

Reply:
oooops - nevermind - I misread the changes. 
relRoot starts relative and is made absolute only if projectDir is absolute. 

I think in this case you would actually want to just check if the projectDir is not inside the OF folder then do: 
`relRoot = getOFRoot();` 

( so the opposite of this - https://github.com/openframeworks/projectGenerator/blob/master/ofxProjectGenerator/src/projects/baseProject.cpp#L222 ) 

Sorry for the noise!! 🤦‍♂️

Reply:
yeah this is the best option.

Reply:
cool - will do! 
I just checked and it is currently making projectDir absolute - so then then relRoot gets made abs. 

```
 relRoot is ../..
 projectDir is "/Users/theo/Documents/CODE/__OPENFRAMEWORKS/openFrameworks/apps/test123"
 relRoot is /Users/theo/Documents/CODE/__OPENFRAMEWORKS/openFrameworks
```

Reply:
yeah with your path inside path detection it will do it right
yesterday before your PR I was searching for the same thing. I was wondering if std::filesystem had a function to check if a path contains another, so nothing in this case. 
I was going to test this snipped but I've seen you already solved in your PR
Maybe we could have a function in Utils.cpp (or the core) so your solution can be reused if needed in other places.

```c++
bool is_subpath(const std::filesystem::path &path,
                const std::filesystem::path &base)
{
    auto rel = std::filesystem::relative(path, base);
    return !rel.empty() && rel.native()[0] != '.';
}
```

Reply:
@dimitre yes! great idea. having a `is_subpath` or even more specific `isPathInOFRoot` or something would be helpful - the rfind approach is kind of hard to follow ( though it seems to work well )